src/Service/Indexing/IndexingElasticService.php

```diff
@@ -0,0 +1,253 @@
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove(int $id): void
+    {
+        $params = [
+            'index' => $this->indexAliasName,
+            'id' => $id,
+        ];
+
+        try {
+            $response = $this->getClient()->delete($params);
+            $this->refreshIndex($this->indexAliasName);
+        } catch (SearchIndexException|ClientResponseException|MissingParameterException|ServerResponseException $e) {
+            throw new SearchIndexException($e->getMessage(), (int) $e->getCode());
+        }
+
+        if (200 !== $response->getStatusCode()) {
+            throw new SearchIndexException('Unable to create new index', $response->getStatusCode());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function bulkAdd(array $items): void
+    {
+        if (!isset($this->newIndexName)) {
+            $this->newIndexName = $this->indexAliasName.'_'.date('Y-m-d-His');
+            $this->createIndex($this->newIndexName);
+        }
+
+        $params = [];
+        foreach ($items as $item) {
+            /* @var IndexItem $item */
+            $params['body'][] = [
+                'index' => [
+                    '_index' => $this->newIndexName,
+                    '_id' => $item->getId(),
+                ],
+            ];
+
+            $params['body'][] = $item->toArray();
+        }
+
+        try {
+            $this->getClient()->bulk($params);
+        } catch (SearchIndexException|ClientResponseException|ServerResponseException $e) {
+            throw new SearchIndexException($e->getMessage(), (int) $e->getCode());
+        }
+    }
+
+    /**
+     * Switch new index with old by updating alias.
+     *
+     * @throws SearchIndexException
+     */
+    public function switchIndex(): void
+    {
+        $existingIndexName = $this->getCurrentActiveIndexName();
+        $this->refreshIndex($this->newIndexName);
+
+        try {
+            $client = $this->getClient();
+            $client->indices()->updateAliases([
+                'body' => [
+                    'actions' => [
+                        [
+                            'add' => [
+                                'index' => $this->newIndexName,
+                                'alias' => $this->indexAliasName,
+                            ],
+                        ],
+                    ],
+                ],
+            ]);
+            $client->indices()->delete(['index' => $existingIndexName]);
+        } catch (ClientResponseException|MissingParameterException|ServerResponseException $e) {
+            throw new SearchIndexException($e->getMessage(), (int) $e->getCode());
+        }
+    }
+
+    /**
+     * Get client to communicate with ES.
+     *
+     * @return client
+     *   ElasticSearch client
+     *
+     * @throws SearchIndexException
+     */
+    private function getClient(): Client
```

Comment:
You can configure the `Client` as a service and just inject it:
```
parameters:
    es_config: {'hosts': ['http://es:9200']}

services:
    # lots of service definitions in between    

    Elasticsearch\ClientBuilder: ~

    Elasticsearch\Client:
        factory:   ['@Elasticsearch\ClientBuilder', fromConfig]
        arguments: ['%es_config%']
```
(See: 
- https://stefan-poeltl.medium.com/symfony-meets-elasticsearch-implement-a-search-as-you-type-feature-307e2244f078
- https://github.com/elastic/elasticsearch-php/blob/e55c738d2592d5a121a93f6c3fdc41af5984925b/src/ClientBuilder.php#L156 )

Reply:
Fixed